File: EventsDetailFragment/app/src/main/java/com/example/eventsdetailfragment/Fragments/PriceDetailsFragment.java
package com.example.eventsdetailfragment.Fragments;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.eventsdetailfragment.FragmentCommunication;
import com.example.eventsdetailfragment.Model.EventModel;
import com.example.eventsdetailfragment.PreviewDetails;
import com.example.eventsdetailfragment.R;

public class PriceDetailsFragment extends Fragment {

    private FragmentCommunication communication;
    private EditText etCurrency, etPrice;
    private Button btnPreview;
    private String title, description, startDate, endDate, startTime, endTime;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_price_details, container, false);
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            title = getArguments().getString("eventTitle");
            description = getArguments().getString("eventDesc");
            startDate = getArguments().getString("startDate");
            endDate = getArguments().getString("endDate");
            startTime = getArguments().getString("startTime");
            endTime = getArguments().getString("endTime");
        }
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initViews(view);
    }

    private void initViews(View view) {
        etCurrency = view.findViewById(R.id.etCurrency);
        etPrice = view.findViewById(R.id.etPrice);
        btnPreview = view.findViewById(R.id.btnPreview);

        btnPreview.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String price = etPrice.getText().toString();
                EventModel eventModel = new EventModel(title, description, startDate, endDate, startTime, endTime, price);
                Intent intent = new Intent(getActivity(), PreviewDetails.class);
                intent.putExtra("eventModel", eventModel);
                startActivity(intent);
            }
        });
    }
}

File: Kotlin Projects/EventsNavFragments/app/src/main/java/com/example/events_navfragments/PriceDetailsFragment.kt
package com.example.events_navfragments

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import kotlinx.android.synthetic.main.fragment_price_details.*


class PriceDetailsFragment : Fragment(R.layout.fragment_price_details) {

    private lateinit var eventName: String
    private lateinit var eventDecs: String
    private lateinit var startTime: String
    private lateinit var endTime: String
    private lateinit var startDate: String
    private lateinit var endDate: String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        getDetailsAndTime()
    }

    private fun getDetailsAndTime() {
        arguments?.run {
            eventName = getString("eventName").toString()
            eventDecs = getString("eventDecs").toString()
            startTime = getString("startTime").toString()
            endTime = getString("endTime").toString()
            startDate = getString("startDate").toString()
            endDate = getString("endDate").toString()

        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        btnPreview.setOnClickListener {
            val currencyAmount = etCurrency.text.toString()
            val price = etPrice.text.toString().toInt()
            val intent = Intent(context, PreviewActivity::class.java)

            intent.putExtra("eventName", eventName)
            intent.putExtra("eventDecs", eventDecs)
            intent.putExtra("startTime", startTime)
            intent.putExtra("endTime", endTime)
            intent.putExtra("startDate", startDate)
            intent.putExtra("endDate", endDate)
            intent.putExtra("price", price)

            startActivity(intent)
        }
    }
}